### PR TITLE
Add build_id to the XRefService Reply messages

### PR DESCRIPTION
**Pull request by danielmoy-google**
Explicitly return a unique identifier for the underlying dataset of the server.
This allows callers to know the state for each individual request, which may
differ from machine to machine in a distributed, productionized environment.
StatusService.Status gives good base info, but runs into eventual consistency
problems during data updates to a distributed system.

### CODE REVIEW

**Comment by jaysachs** [comment]
> Explicitly returning a build_id indicating the state of the underlying build.
> This allows callers to know the state for each individual request, which may
> differ from machine to machine in a distributed, productionized environment.
> StatusService.Status gives good base info, but runs into eventual consistency
> problems during data updates to a distributed system.

This description doesn't seem to match with the new field. It's not the "state" of the underlying build, it's an identifier for the build (that could be used to join against some external data sources to determine the state).

**Comment by danielmoy-google** [comment]
> This description doesn't seem to match with the new field. It's not the "state" of the underlying build, it's an identifier for the build (that could be used to join against some external data sources to determine the state).

Okay, updated description

File: kythe/proto/xref.proto [hunk 1]
```diff
@@ -265,6 +265,9 @@ message DecorationsReply {
   // references to the list of their overrides.
   map<string, Overrides> extends_overrides = 17;
 
+  // A unique build_id for the underlying dataset serving this reply.
```
Comment (jaysachs) [inline kythe/proto/xref.proto:268]: s/build_id/identifier/ in the doc comments here and elsewhere.

**Review by jaysachs — APPROVED** [review]

File: kythe/proto/xref.proto (hunk 1 above)
Comment (danielmoy-google) [reply kythe/proto/xref.proto]: done

**Pull request merged**